pages/lot/[id].js

```diff
@@ -208,10 +212,13 @@ function Permits({ data }) {
 }
 
 export default function Lot({ id }) {
-  const router = useRouter();
   const [data, isLoading] = useLotData(id);
+  const [lotNumber, setLotNumber] = useState(null);
+  const { colorMode } = useColorMode();
 
-  const lotNumber = parcelNumberToLotNumber(router.query.id);
+  useEffect(() => {
+    setLotNumber(parcelNumberToLotNumber(id));
```

Comment:
moved this here because it was running before component mount sometimes, so was returning undefined when slicing

Reply:
👍